File: iosApp/iosApp/feature/main/MainScreenEvent.swift
//
//  MainScreenEvent.swift
//  iosApp
//
//  Created by Mikhail Maevskii on 15/12/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import Shared

enum MainScreenEvent {
    case NextMonthBtnClicked
    case PreviousMonthBtnClicked
    case CalendarViewSwitch
    case CalendarDayClicked(dayIndex: Int)
    case EditTaskClicked(task: Task)
    case DeleteTaskClicked(task: Task)
    case TaskStatusChangeClicked(task: Task)
    case MessageDialogDismissed
    case EditTaskBottomSheetDismissed
    case SaveUpdatedTaskClicked(updatedTask: Task)
    case RecordNewTaskBtnClicked
    case NewTaskRecordingFinished(recognizedRequest: String)
    case NewTaskRecordingDismissed
    case RecognizedTaskTextEdited(updatedText: String)
    case CreateNewTaskClicked
}

File: composeApp/src/androidMain/kotlin/dem/dev/timeflame/feature/main/MainScreenEvent.kt
package dem.dev.timeflame.feature.main

import dem.dev.timeflame.domain.model.Task
import dem.dev.timeflame.feature.calendar.model.CalendarDay

sealed interface MainScreenEvent {
    data object NextMonthBtnClicked: MainScreenEvent
    data object PreviousMonthBtnClicked: MainScreenEvent
    data object CalendarViewSwitch: MainScreenEvent
    data class CalendarDayClicked(val calendarDayIndex: Int): MainScreenEvent
    data class EditTaskClicked(val task: Task): MainScreenEvent
    data class DeleteTaskClicked(val task: Task): MainScreenEvent
    data class TaskStatusChangeClicked(val task: Task): MainScreenEvent
    data object MessageDialogDismissed: MainScreenEvent
    data class SaveUpdatedTaskBtnClicked(val updatedTask: Task): MainScreenEvent
    data object EditTaskBottomSheetDismissed: MainScreenEvent
    data object RecordNewTaskBtnClicked: MainScreenEvent
    data class NewTaskRecordingFinished(val recognizedRequest: String): MainScreenEvent
    data object NewTaskRecordingDismissed: MainScreenEvent
    data class RecognizedTaskTextEdited(val updatedText: String): MainScreenEvent
    data object CreateNewTaskClicked: MainScreenEvent
    data object ProfileContextMenuClicked: MainScreenEvent
    data object ProfileContextMenuDismissed: MainScreenEvent
    data object LogoutBtnClicked: MainScreenEvent
}